src/components/button/button.scss

```diff
@@ -655,15 +656,26 @@
   --calcite-button-padding-x: 15px;
   @apply text-0h;
 }
-:host([scale="l"]:not([appearance="transparent"])) button.content--slotted,
-:host([scale="l"]:not([appearance="transparent"])) a.content--slotted {
-  --calcite-button-padding-y: 11px;
+
+:host([scale="l"]:not([appearance="transparent"])) {
+  .button-padding {
+    --calcite-button-padding-y: 11px;
+  }
+  //shrink the padding if icon is present to preserve the height
+  .button-padding-shrunk {
+    --calcite-button-padding-y: 9px;
+  }
 }
 // accommodate for transparent buttons not having borders
-:host([scale="l"][appearance="transparent"]) button.content--slotted,
-:host([scale="l"][appearance="transparent"]) a.content--slotted {
-  --calcite-button-padding-x: theme("padding.4");
-  --calcite-button-padding-y: theme("padding.3");
+:host([scale="l"][appearance="transparent"]) {
+  .button-no-border-padding {
```

Comment:
nitpick: I think the selector for this ideally would be `button--no-border-padding` using BEM naming methodology.  https://getbem.com/